diffkemp/semdiff/module_diff.py

```diff
@@ -35,9 +97,27 @@ def modules_diff(first, second, param, timeout, function, syntax_only=False,
                 stat.log_result(Result.ERROR, "")
                 return stat
 
+        if param.indices is not None:
+            # Do not compare function when all uses are of a part of the
+            # variable with a different index than specified in the indices
+            # parameter.
+            llvm_context = get_global_context()
+            module_left = llvm_context.parse_ir(
+                create_memory_buffer_with_contents_of_file(first.llvm))
+            module_right = llvm_context.parse_ir(
+                create_memory_buffer_with_contents_of_file(second.llvm))
+            function_left = module_left.get_named_function(c.first)
+            function_right = module_right.get_named_function(c.second)
+
+            if (_corresponding_index_unused(module_left, function_left,
+                                            param) and
+                _corresponding_index_unused(module_right, function_right,
+                                            param)):
+                continue
+
```

Comment:
All this along with the above functions (_corresponding_index_unused, _indices_correspond) should be a part of FunctionCouplings.
Change the method infer_for_param to take a KernelParam and check for indices, too.